pkg/componentreadiness/jiraautomator/jiraautomator.go

```diff
@@ -0,0 +1,597 @@
+	values := url.Values{}
+	if reportOpts.TestIDOption.TestID != "" {
+		absURL += "env_test?"
+		values.Add("testId", reportOpts.TestIDOption.TestID)
+		values.Add("capability", reportOpts.TestIDOption.Capability)
+		values.Add("component", reportOpts.TestIDOption.Component)
+	} else if reportOpts.TestIDOption.Capability != "" {
+		absURL += "env_capability?"
+		values.Add("capability", reportOpts.TestIDOption.Capability)
+		values.Add("component", reportOpts.TestIDOption.Component)
+	} else if reportOpts.TestIDOption.Component != "" {
+		absURL += "env_capabilities?"
+		values.Add("component", reportOpts.TestIDOption.Component)
+	} else {
+		absURL += "main?"
+	}
+
+	if reportOpts.BaseRelease.Release != "" {
+		values.Add("baseRelease", reportOpts.BaseRelease.Release)
+		values.Add("baseStartTime", reportOpts.BaseRelease.Start.UTC().Format(time.RFC3339))
+		values.Add("baseEndTime", reportOpts.BaseRelease.End.UTC().Format(time.RFC3339))
+	}
+	if reportOpts.SampleRelease.Release != "" {
+		values.Add("sampleRelease", reportOpts.SampleRelease.Release)
+		values.Add("sampleStartTime", reportOpts.SampleRelease.Start.UTC().Format(time.RFC3339))
+		values.Add("sampleEndTime", reportOpts.SampleRelease.End.UTC().Format(time.RFC3339))
+	}
+	values.Add("columnGroupBy", strings.Join(reportOpts.VariantOption.ColumnGroupBy.List(), ","))
+	for name, variants := range reportOpts.VariantOption.IncludeVariants {
+		for _, v := range variants {
+			values.Add("includeVariant", fmt.Sprintf("%s:%s", name, v))
+		}
+	}
+
+	values.Add("confidence", fmt.Sprintf("%d", reportOpts.AdvancedOption.Confidence))
+	if reportOpts.AdvancedOption.IgnoreDisruption {
+		values.Add("ignoreDisruption", "1")
+	} else {
+		values.Add("ignoreDisruption", "0")
+	}
+	if reportOpts.AdvancedOption.IgnoreMissing {
+		values.Add("ignoreMissing", "1")
+	} else {
+		values.Add("ignoreMissing", "0")
+	}
+	values.Add("minFail", fmt.Sprintf("%d", reportOpts.AdvancedOption.MinimumFailure))
+	values.Add("pity", fmt.Sprintf("%d", reportOpts.AdvancedOption.PityFactor))
+
+	absURL += values.Encode()
+	viewURL := j.sippyURL + "/sippy-ng/component_readiness/main?view=" + view.Name
+	return absURL, viewURL, nil
+}
+
+// createNewJiraIssueForRegressions creates new issue for components by
+// a. setting the ticket's Affects Version/s= sample version.
+// b  adding the Regression label defined by LabelJiraAutomator.
+// c. setting a description with links to CR
+// d. for pre-release, setting "Release Blocker" label to Approved
+func (j JiraAutomator) createNewJiraIssueForRegressions(view crtype.View, component string, tests []crtype.ReportTestSummary, linkedIssue *jira.Issue) (*jira.Issue, error) {
+	if len(tests) > 0 {
+		description := `Component Readiness has found a potential regression in the following tests:`
+		for i, test := range tests {
+			// Only show stats for the worst regression
+			if i == 0 {
+				description += fmt.Sprintf("\n h4. Most Regressed Test:\n{code}%s{code}\n", test.TestName)
```

Comment:
I still worry that strongly separating out a most regressed test sends the wrong message, and could lead to someone focusing on one issue and ignoring the others. From our perspective we really don't care, they're all equal in my eyes. 

I can't remember what we decided when this was discussed, if we decided to go ahead with this then ignore me. :)

Reply:
Will keep all tests on the same level.